server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java

```diff
@@ -0,0 +1,142 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.coffeehouse.servlets;
+
+import com.google.coffeehouse.common.Person;
+import com.google.coffeehouse.storagehandler.StorageHandlerApi;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/api/update-person")
+public class UpdatePersonServlet extends HttpServlet {
+  /** The fields of the Person object that can be updated. */
+  public static final List<String> updateableFields = Arrays.asList(
+      Person.NICKNAME_FIELD_NAME, Person.EMAIL_FIELD_NAME, Person.PRONOUNS_FIELD_NAME);
+  /** 
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request cannot be parsed for any reason.
+   */
+  public static final String BODY_ERROR = "- unable to parse body.";
+  /** Message to be logged when the body of the POST request cannot be parsed. */
+  public static final String LOG_BODY_ERROR_MESSAGE =
+      "Body unable to be parsed in UpdatePersonServlet: ";
+  /** 
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request does not have a "person" field.
+   */
+  public static final String NO_PERSON_ERROR = "No \"person\" found in JSON.";
+  /** 
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request does not have a "userId" field in its "person" field.
+   */
+  public static final String NO_USER_ID_ERROR = "No \"userId\" found in JSON.";
+  /** Message to be logged when the body of the POST request does not have required fields. */
+  public static final String LOG_INPUT_ERROR_MESSAGE =
+      "Error with JSON input in UpdatePersonServlet: ";
+  /** Name of the key in the input JSON that corresponds to the person object. */
+  public static final String PERSON_FIELD_NAME = "person";
+  /** Name of the key in the input JSON that corresponds to the update mask. */
+  public static final String UPDATE_MASK_FIELD_NAME = "updateMask";
+
+  private static final Gson gson = new Gson();
+  private static StorageHandlerApi storageHandler;
+
+  /** 
+   * Overloaded constructor for dependency injection.
+   * @param storageHandler the {@link StorageHandlerApi} that is used when fetching the Person
+   */
+  public UpdatePersonServlet(StorageHandlerApi storageHandler) {
+    super();
+    this.storageHandler = storageHandler;
+  }
+
+  /** 
+   * Explicit default constructor used for instantiating the servlet when not testing.
+   */
+  public UpdatePersonServlet() {
+    super();
+    this.storageHandler = new StorageHandlerApi();
+  }
```

Comment:
I'm not sure we want this constructor. Shouldn't there only be one instance of StorageHandlerApi in the app?

Reply:
All StorageHandlerApi does is call static methods on StorageHandlerApi with a private static final dbClient that it has already created. The reason it was passed in to the constructor was solely to mock it easier (previously, we just referenced it statically like `StorageHandlerApi.fetchPersonFromId()`). Being honest I'm not sure how much of a problem this is, if it is a big problem, then we can reference it statically instead, I just don't know how to mock that out in unit tests.

Reply:
Hmm, ok. Seems like kind of an odd pattern but I can't really think of anything better just given the way Servlets work so, SG